lib/javascripts/base_app.js

```diff
@@ -55,17 +48,26 @@ function BaseApp(zafClient, data) {
   this.I18n = { t: I18n.t };
   registerHelpers(this);
   bindEvents(this);
-  let evt = { firstLoad: true };
+
   this._metadata = data.metadata;
   this._context = data.context;
-  zafClient.get('currentUser.locale').then(function(data) {
-    I18n.loadTranslations(data['currentUser.locale']);
-    if (this.defaultState) {
-      this.switchTo(this.defaultState);
-    }
-    resolveHandler(this, 'app.created')();
-    resolveHandler(this, 'app.activated')(evt, evt);
-  }.bind(this));
+  this._storage = new Storage(this._metadata.installationId);
+
+  let view = new View({ onRender: () => {
```

Comment:
I think that would be one approach, not sure about typical. I started to dislike mixins after working with Ember. I think it's too easy to misuse mixins, for example you can easily write a mixin that relies on a specific class to be mixed into and doesn't serve any other purpose. I was going with a composition based approach. Perhaps a better approach here would be emitting an event on render, but it would require a dependency on an event emitter library. For this use case alone, a callback seemed pretty standard.